pysph/sph/wc/crksph.py

```diff
@@ -0,0 +1,164 @@
+from pysph.sph.equation import Equation
+
+
+class CRKSPHPreStep(Equation):
+
+    def __init__(self, dest, sources, dim=2):
+        self.dim = dim
+        super(CRKSPHPreStep, self).__init__(dest, sources)
+
+    def loop_all(self, d_idx, d_x, d_y, d_z, d_h, s_x, s_y, s_z, s_h, s_m,
+                 s_rho, KERNEL, NBRS, N_NBRS, d_ai, d_gradai, d_bi, d_gradbi):
+        x = d_x[d_idx]
+        y = d_y[d_idx]
+        z = d_z[d_idx]
+        h = d_h[d_idx]
+        i, j, k, s_idx, d = declare('int', 5)
+        alp, bet, gam, phi, psi = declare('int', 5)
+        xij = declare('matrix(3)')
+        dwij = declare('matrix(3)')
+        d = self.dim
+        tol = 1.0e-10
```

Comment:
Why this arbitrary number 1e-10?

Reply:
This is to prevent the denominator to be 0, changed this to 1.0e-12 to make it further small